File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int removed=0,count=0;
        while(!s.equals("1")){
            int sizeAfter=0;
            for(int i=0;i<s.length();i++){
                if(s.charAt(i)=='1') sizeAfter++;
                else removed++;
            }
            count++;
            s=Integer.toBinaryString(sizeAfter);
        }
        return new int[]{count,removed};
    }
}
//filter 0, size 구함, toBinaryString
//size diff를 구해 저장해야 함

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let repetition=0;
    let removed=0;
    while(s!=="1"){
        const prevStr=[...s];
        const nextNum=prevStr.filter(i=>i==='1').length;
        removed+=prevStr.length-nextNum;
        s=nextNum.toString(2);
        repetition++;
    }
    return [repetition, removed];
}